rpc-jobs/multi-node-aio.yml

```diff
@@ -36,77 +49,95 @@
+            dir("openstack-ansible-ops/multi-node-aio"){
+              common.conditionalStage(
+                stage_name: 'Setup Host',
+                stage: {
+                  common.run_script (
+                    script: 'setup-host.sh',
+                    environment_vars: ["PARTITION_HOST=${env.PARTITION_HOST}"]
+                  )
+                } //stage
+              ) //conditionalStage
 
-          common.conditionalStage(
-            stage_name: 'Setup Virtual Networks',
-            stage: {
-              common.run_script (
-                script: 'setup-virsh-net.sh',
-                environment_vars: []
-              )
-            } //stage
-          ) //conditionalStage
+              common.conditionalStage(
+                stage_name: 'Setup Cobbler',
+                stage: {
+                  common.run_script (
+                    script: 'setup-cobbler.sh',
+                    environment_vars: ["DEFAULT_IMAGE=${env.DEFAULT_IMAGE}"]
+                  )
+                } //stage
+              ) //conditionalStage
 
-          common.conditionalStage(
-            stage_name: 'Deploy VMs',
-            stage: {
-              common.run_script (
-                script: 'deploy-vms.sh',
-                environment_vars: []
-              )
-            } //stage
-          ) //conditionalStage
+              common.conditionalStage(
+                stage_name: 'Setup Virtual Networks',
+                stage: {
+                  common.run_script (
+                    script: 'setup-virsh-net.sh',
+                    environment_vars: []
+                  )
+                } //stage
+              ) //conditionalStage
 
-          common.conditionalStage(
-            stage_name: 'Setup OpenStack Ansible',
-            stage: {
-              common.run_script (
-                script: 'deploy-osa.sh',
-                environment_vars: [
-                "OSA_BRANCH=${env.OPENSTACK_ANSIBLE_BRANCH}",
-                "RUN_OSA=false"]
-              )
-            } //stage
-          ) //conditionalStage
-        } // dir
-      } // node
+              common.conditionalStage(
+                stage_name: 'Deploy VMs',
+                stage: {
+                  common.run_script (
+                    script: 'deploy-vms.sh',
+                    environment_vars: []
+                  )
+                } //stage
+              ) //conditionalStage
+
+              common.conditionalStage(
+                stage_name: 'Setup OpenStack Ansible',
+                stage: {
+                  common.run_script (
+                    script: 'deploy-osa.sh',
+                    environment_vars: [
+                    "OSA_BRANCH=${env.OPENSTACK_ANSIBLE_BRANCH}",
+                    "RUN_OSA=false"]
+                  )
+                } //stage
+              ) //conditionalStage
+            } // dir
+          }// public cloud node
+        } catch (Exception e){
```

Comment:
I've been playing with this today.. Exception doesn't catch all throwables (some of them derive from Error instead). But as @mattt416 pointed out, you can just catch(e). 

On that note, if you print(e) you don't get a full stack trace, so it may be better to set the build result then re-throw e.